ddl/scheduler.go

```diff
@@ -182,19 +184,27 @@ func (b *backfillSchedulerHandle) SplitSubtask(_ context.Context, subtask []byte
 		}
 	}
 	ingestScheduler.close(false)
-	// TODO: unsafe import.
+
+	_, err = b.bc.Flush(b.index.ID)
+	if err != nil {
+		logutil.BgLogger().Error("[ddl] flush error", zap.Error(err))
+		return nil, err
+	}
+	err = b.bc.UnsafeImportAndReset(b.d.ctx, b.index.ID, int64(lightning.SplitRegionSize)*int64(lightning.MaxSplitRegionSizeRatio), int64(lightning.SplitRegionKeys))
+	if err != nil {
+		if common.ErrFoundDuplicateKeys.Equal(err) {
+			err = convertToKeyExistsErr(err, b.index, b.ptbl.Meta())
+		}
+		logutil.BgLogger().Error("[ddl] unsafe import and reset error", zap.Error(err))
+		return nil, err
+	}
 	return nil, consumer.getResult()
 }
 
 // CleanupSubtaskExecEnv implements the Scheduler interface.
 func (b *backfillSchedulerHandle) CleanupSubtaskExecEnv(context.Context) error {
 	logutil.BgLogger().Info("[ddl] lightning cleanup subtask exec env")
 
-	err := b.bc.FinishImport(b.index.ID, b.index.Unique, b.ptbl)
```

Comment:
Must call finishImport to close engine and backend correctly before scheduler finished one step of the global task, otherwise, it will still be leak.

Reply:
engine is closed by Unregister()